src/python/pants/goal/run_tracker.py

```diff
@@ -390,3 +413,40 @@ def shutdown_worker_pool(self):
     N.B. This exists only for internal use and to afford for fork()-safe operation in pantsd.
     """
     SubprocPool.shutdown(self._aborted)
+
+  def report_target_info(self, scope, target, keys, val):
+    """Add target information to run_info under target_data.
+
+    Will Recursively construct a nested dict with the keys provided.
```

Comment:
Yea, that looks like a bug. My primary concern is that this usecase works:
```
report_target_info(scope, target, ['this', 'is', 'a', 'deeply', 'namespaced'], 'value1')
report_target_info(scope, target, ['this', 'is', 'another', 'deeply', 'namespaced'], 'value2')
```

If need be, removing the ability to also use:
```
report_target_info(scope, target, ['this'], {'is': {'a': {'deeply': {'namespaced': 'value'}}}})
```
...would be preferable to me.